media-plugins/gst-plugins-v4l2/files/gst-plugins-v4l2-1.22.3-clang16-build-fix.patch

```diff
@@ -0,0 +1,12 @@
+Bug: https://bugs.gentoo.org/896418
+--- a/sys/v4l2/gstv4l2object.c
++++ b/sys/v4l2/gstv4l2object.c
+@@ -544,7 +544,7 @@ gst_v4l2_object_new (GstElement * element,
+     v4l2object->fd_open = NULL;
+     v4l2object->close = close;
+     v4l2object->dup = dup;
+-    v4l2object->ioctl = ioctl;
++    v4l2object->ioctl = (void *)ioctl;
```

Comment:
I don't think it's a fully correct fix.

The main problem is that `ioctl` has different signatures in glibc and musl.

musl:

```c
int ioctl (int, int, ...);
```

glibc:

```c
extern int ioctl (int __fd, unsigned long int __request, ...) __THROW;
```

libv4l just assumes the glibc definition of ioctl:

```c
int v4l2_ioctl(int fd, unsigned long int request, ...)
```

https://git.linuxtv.org/v4l-utils.git/tree/lib/libv4l2/libv4l2.c?id=fce3922fbe38fb005bb864d8780d67735be1a3d1#n1054

They have guards for the ioctl definitions (POSIX vs GNU) in other places, like:

https://git.linuxtv.org/v4l-utils.git/tree/lib/libv4l2/v4l2convert.c?id=fce3922fbe38fb005bb864d8780d67735be1a3d1#n126

So I think that the correct fix would be to add these guards to `libv4l2.c` as well. I will do that now.